nixos/tests/matrix/synapse-workers.nix

```diff
@@ -0,0 +1,45 @@
+  import ../make-test-python.nix ({ pkgs, ... }:
+  {
+    name = "matrix-synapse-workers";
+    meta = {
+    maintainers = pkgs.matrix-synapse.meta.maintainers;
+    };
+
+    nodes = {
+      homeserver = { pkgs, nodes, ... }: {
+        services.postgresql = {
+          enable = true;
+          initialScript = pkgs.writeText "synapse-init.sql" ''
+            CREATE ROLE "matrix-synapse" WITH LOGIN PASSWORD 'synapse';
+            CREATE DATABASE "matrix-synapse" WITH OWNER "matrix-synapse"
+            TEMPLATE template0
+            LC_COLLATE = "C"
+            LC_CTYPE = "C";
+          '';
+        };
+
+        services.redis.servers."".enable = true;
+
+        services.matrix-synapse = {
+          enable = true;
+          settings = {
+            enable_registration = true;
+            enable_registration_without_verification = true;
```

Comment:
We should test verification, too.

Reply:
This has nothing to do with this PR.
There's a test for normal "synapse" that actually tests registration. Arguably as this test is written we don't need to set these settings at all